packages/terra-tabs/src/common-tabs/_Tabs.jsx

```diff
@@ -336,30 +385,83 @@ class Tabs extends React.Component {
 
   wrapOnSelect(onSelect) {
     return (itemKey, metaData) => {
-      this.setIsOpen(false);
-      onSelect(itemKey, metaData);
+      const updatedTabData = this.state.visibleTabData.map((tab) => ({
+        ...tab,
+        isSelected: tab.itemKey === itemKey,
+      }));
+      this.setState({ visibleTabData: updatedTabData }, () => {
+        this.setIsOpen(false);
+        onSelect(itemKey, metaData);
+      });
     };
   }
 
   wrapOnSelectHidden(onSelect) {
     return (itemKey, metaData) => {
       if (this.isOpen) {
-        onSelect(itemKey, metaData);
+        const updatedTabData = this.state.visibleTabData.map((tab) => ({
+          ...tab,
+          isSelected: tab.id === itemKey,
+        }));
+        this.setState({ visibleTabData: updatedTabData }, () => {
+          onSelect(itemKey, metaData);
+          this.setIsOpen(!this.isOpen);
+        });
       }
-      this.setIsOpen(!this.isOpen);
     };
   }
 
+  wrapOnClose() {
+    return (itemKey, event) => {
+      this.addButtonToggle = false;
+      this.setIsOpen(false);
+      let removedTabIndex = -1;
+      const updatedTabData = this.state.visibleTabData
+        .map((tab, index) => {
+          if (tab.itemKey === itemKey && tab.isSelected === true) {
+            removedTabIndex = index;
+          }
+          return { ...tab, originalIndex: index };
+        })
+        .filter((tab) => tab.itemKey !== itemKey);
+
+      if (!updatedTabData.some((tab) => tab.isSelected === true) && updatedTabData.length > 0 && removedTabIndex !== 0 && updatedTabData[removedTabIndex - 1] !== undefined && updatedTabData[removedTabIndex - 1].isDisabled !== true) {
+        if (updatedTabData[removedTabIndex - 1].isSelected !== undefined) {
+          updatedTabData[removedTabIndex - 1].isSelected = true;
+        }
+      } else if (!updatedTabData.some((tab) => tab.isSelected === true) && updatedTabData.length > 0 && removedTabIndex !== 0 && updatedTabData[removedTabIndex - 1] !== undefined && updatedTabData[removedTabIndex - 1].isDisabled === true) {
+        if (updatedTabData[removedTabIndex - 2]?.isSelected !== undefined) {
+          updatedTabData[removedTabIndex - 2].isSelected = true;
+        }
+      } else if (removedTabIndex >= 0 && removedTabIndex < updatedTabData.length) {
+        if (updatedTabData[removedTabIndex]?.isSelected !== undefined) {
+          updatedTabData[removedTabIndex].isSelected = true;
+        }
+      }
+      this.props.onTabStateChange(updatedTabData, itemKey, event);
+      this.setState({ visibleTabData: updatedTabData });
+    };
+  }
+
+  wrapOnAddButton() {
+    if (this.props.onSelectAddButton) {
+      this.addButtonToggle = true;
```

Comment:
This toggle is used to keep focus on newly created tab ,after AddTab button click.

Reply:
I thought we decided to remove this variable and to use the existing Ref instead of variable. https://github.com/cerner/terra-framework/pull/1738#discussion_r1297175469